Feat/display a warning when using an unauthorized email address - MAILPOET-4245

triple0t opened the pull request:
[MAILPOET-4245](https://mailpoet.atlassian.net/browse/MAILPOET-4245)

**Testing**:

* Activate MSS sending method
* Visit Newsletter sending page
* If your sender's email address is authorized, you shouldn't get any message otherwise you would receive a message to authorized your email address
* Update your sender email address, check the message.
* Test with both authorized and unauthorized email addresses


<img width="478" alt="Screenshot 2022-07-01 at 3 25 33 PM" src="https://user-images.githubusercontent.com/30554163/176923336-e35026c6-3cd0-4d7e-af60-d4d3ac65ad03.png">

triple0t committed:
Add authorized_emails to newsletters data

MAILPOET-4245

triple0t committed:
Verify authorized email address on newsletter sending page

MAILPOET-4245

triple0t committed:
Fix issues with CSS and https://user-images.githubusercontent.com/30554163/176655015-777259ac-5748-48b0-8f27-41befac015a9.png

MAILPOET-4245

triple0t committed:
Reuse authorized email address message

MAILPOET-4245

lysyjan commented on mailpoet/assets/js/src/common/sender_email_address_warning.jsx:
```diff
@@ -5,8 +5,33 @@ import ReactStringReplace from 'react-string-replace';
+          )}{' '}
+          <a
+            className="mailpoet-link"
+            href={`https://account.mailpoet.com/authorization?email=${emailAddress}`}
```
We have [a URL factory](https://github.com/mailpoet/mailpoet/blob/trunk/mailpoet/assets/js/src/mailpoet_com_url_factory.js) for similar purposes. Maybe we can add a method for this new URL type. What do you think about that?

triple0t replied:
That's definitely a very good suggestion. 
We will be changing the behavior of that link in this ticket https://mailpoet.atlassian.net/browse/MAILPOET-4300
We won't require linking to the Shop anymore 
So I thought to just keep it as it is for now

lysyjan commented on mailpoet/assets/js/src/newsletters/send/sender_address_field.jsx:
```diff
@@ -49,7 +49,7 @@ class SenderField extends Component {
           onValueChange={this.onChange}
         />
-        <div className="regular-text">
+        <div className="regular-text" style={{ marginTop: '2.5rem' }}>
```
Is this margin necessary? It looks better without it to me.

triple0t replied:
Yeah, but that's to fix this issue https://user-images.githubusercontent.com/30554163/176655015-777259ac-5748-48b0-8f27-41befac015a9.png

triple0t commented:
Hi @lysyjan 

Thanks for the review.

Please check my comments [here](https://github.com/mailpoet/mailpoet/pull/4216#discussion_r911789593) and [here](https://github.com/mailpoet/mailpoet/pull/4216#discussion_r911785225) 

Do let me know if I should make any more updates.

lysyjan commented:
@triple0t Thank you for the quick reactions. I will move the PR forward. Please update the image in the description because it's outdated. You can avoid future questions from QA.

veljkho reviewed (changes requested):
It is good, but one thing to ask.. Can we make it the same as it works in MailPoet > Settings > Basic tab? The difference is that when I start writing an email it immediately tells me that the email is not authorized.. until the email is fully written then the message disappears (in case it's authorized email).
I would expect when I finish writing email and then changing the cursor outside of the field to see if it is authorized or not.. that's how it works in the settings now.

triple0t committed:
Add onBlurEvent, show warning message after typing

Add onBlurEvent to the FormFieldText component and the sender address field

This would update the validation to show the warning message only after the user is done typing

MAILPOET-4245

triple0t commented:
> It is good, but one thing to ask.. Can we make it the same as it works in MailPoet > Settings > Basic tab? The difference is that when I start writing an email it immediately tells me that the email is not authorized.. until the email is fully written then the message disappears (in case it's authorized email).
> I would expect when I finish writing email and then changing the cursor outside of the field to see if it is authorized or not.. that's how it works in the settings now.

Thanks for the review and suggestions @veljkho 

That component does not offer that feature, but I have updated that in https://github.com/mailpoet/mailpoet/pull/4216/commits/77d6a3ad657535293fb7a41494baa07d641df61a
